### PR TITLE
Functions for integration into semantic-segmentation-editor

### PR DESCRIPTION
**Pull request by pmokeev**

### CODE REVIEW

File: tests/test_compression_algorithms.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    payload = LZW.compress(start_string)
+    payload = FIC.compress(payload)
+
+    data = FIC.decompress(payload)
+    end_string = LZW.decompress(data)
+
+    assert start_string == end_string
+
+
+def test_compress_decompress_text_from_file():
```
Comment (DmiitriyJarosh) [inline tests/test_compression_algorithms.py:35]: Do you have some ground truth compressed texts? maybe from online converter? We can use them to check methods not with each other but with gt too

Comment (pmokeev) [reply tests/test_compression_algorithms.py:35]: Do you mean add a test with the text of some external file? Probably I can add a file with labels from semantic-segmentation-editor.

File: src/algorithmForNN/fileUtils.py [hunk 2]
```diff
@@ -12,17 +12,36 @@ def get_point_cloud_from_bin_file(path_to_bin_file: str) -> o3d.geometry.PointCl
...
+    labels_string = ""
+    for item in labels_list:
+        labels_string += item
+
+    with open(path_to_new_label_file, "wb") as label_file, open(
+        path_to_new_object_file, "wb"
+    ) as object_file:
+
+        compresses_labels_string = FIC.compress(LZW.compress(labels_string))
+        compressed_objects_string = FIC.compress(LZW.compress(objects_string))
```
[the remark this answers is not in the document]
Comment (pmokeev) [reply src/algorithmForNN/fileUtils.py:40]: I have not fully figured it out. Having tried the tool on my own, I got empty objects files

File: tests/data/text.txt [hunk 3]
```diff
@@ -0,0 +1,42 @@
+I don't know what to write, so I'll write a recipe for stuffed cabbage.
```
Comment (anastasiia-kornilova) [inline tests/data/text.txt:1]: Wtf?!

File: tests/test_compression_algorithms.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+def test_compress_decompress_FIC_list():
+    start_list = [1, 4, 3, 250, 4056, 10, 128, 1024, 253, 13, 53]
+    end_list = FIC.decompress(FIC.compress(start_list))
+
+    assert start_list == end_list
+
+
+def test_full_compress_decompress_simple_string():
+    start_string = "Skoltech is the best"
+    payload = LZW.compress(start_string)
```
Comment (anastasiia-kornilova) [inline tests/test_compression_algorithms.py:26]: Use one-line call for those two lines

File: tests/test_compression_algorithms.py (hunk 1 above)
Comment (anastasiia-kornilova) [reply tests/test_compression_algorithms.py:35]: Yes, let's add. In our case we should follow mainly not original impl but one from the editor, so let's add tests for checking equivalence of your impl and original.

**Review by anastasiia-kornilova — CHANGES_REQUESTED** [review]

File: src/algorithmForNN/fileUtils.py [hunk 5]
```diff
@@ -12,17 +13,34 @@ def get_point_cloud_from_bin_file(path_to_bin_file: str) -> o3d.geometry.PointCl
...
+    for item in labels_list:
+        labels_string += item
+
+    with open(path_to_new_label_file, "wb") as label_file, open(
+        path_to_new_object_file, "wb"
+    ) as object_file:
+
+        compresses_labels_string = FIC.compress(LZW.compress(labels_string))
+        compressed_objects_string = FIC.compress(LZW.compress("[]"))
+        for byte in compresses_labels_string:
```
Comment (anastasiia-kornilova) [inline src/algorithmForNN/fileUtils.py:40]: Do we need this iteratior? Why not just smth like https://stackoverflow.com/a/18367068

File: tests/test_compression_algorithms.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    end_string = LZW.decompress(
+        FIC.decompress(FIC.compress(LZW.compress(start_string)))
+    )
+
+    assert start_string == end_string
+
+
+def test_compress_decompress_text_from_file():
+    path_to_correct_file = "tests/data/correct.pcd.labels"
+    path_to_test_file = "tests/data/ownData.txt"
```
Comment (anastasiia-kornilova) [inline tests/test_compression_algorithms.py:35]: Not clear whar is "ownData", so, rename or add comments, please

File: tests/test_compression_algorithms.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+def test_full_compress_decompress_simple_string():
+    start_string = "Skoltech is the best"
+    end_string = LZW.decompress(
+        FIC.decompress(FIC.compress(LZW.compress(start_string)))
+    )
+
+    assert start_string == end_string
+
+
+def test_compress_decompress_text_from_file():
```
Comment (anastasiia-kornilova) [inline tests/test_compression_algorithms.py:33]: Name this method better, as I understood it is connected to compatibility between resulted files, so show this in the name

File: src/algorithmForNN/fileUtils.py [hunk 8]
```diff
@@ -12,17 +13,34 @@ def get_point_cloud_from_bin_file(path_to_bin_file: str) -> o3d.geometry.PointCl
...
+            label_file.write(byte)
+        for byte in compressed_objects_string:
```
Comment (anastasiia-kornilova) [inline src/algorithmForNN/fileUtils.py:42]: The same here

File: src/algorithmForNN/fileUtils.py [hunk 9]
```diff
@@ -12,17 +13,34 @@ def get_point_cloud_from_bin_file(path_to_bin_file: str) -> o3d.geometry.PointCl
...
-    list_strings = ["0\n"] * point_cloud_size
+    labels_list = ["["] + ["0,"] * point_cloud_size + ["]"]
     for index in index_list:
-        list_strings[index] = "1\n"
-    with open(path_to_new_label_file, "w") as file:
-        file.writelines(list_strings)
+        labels_list[index + 1] = "1,"
+    labels_list[-2] = labels_list[-2][:-1]
+
+    labels_string = ""
```
Comment (anastasiia-kornilova) [inline src/algorithmForNN/fileUtils.py:30]: Why not simply `join`? Also, read about list comprehension in Python. Not useful here, but it should be "red flag" for you when you start writing "for" loop.

File: src/algorithmForNN/fileUtils.py [hunk 10]
```diff
@@ -12,17 +13,34 @@ def get_point_cloud_from_bin_file(path_to_bin_file: str) -> o3d.geometry.PointCl
...
     """
     Function changes those lines in the file, the point with the index of which belongs to any plane
     """
-    list_strings = ["0\n"] * point_cloud_size
+    labels_list = ["["] + ["0,"] * point_cloud_size + ["]"]
     for index in index_list:
-        list_strings[index] = "1\n"
-    with open(path_to_new_label_file, "w") as file:
-        file.writelines(list_strings)
+        labels_list[index + 1] = "1,"
```
Comment (anastasiia-kornilova) [inline src/algorithmForNN/fileUtils.py:27]: List comprehension?

**Review by anastasiia-kornilova — CHANGES_REQUESTED** [review]

File: src/algorithmForNN/fileUtils.py (hunk 10 above)
Comment (pmokeev) [reply src/algorithmForNN/fileUtils.py]: I don't understand what this is about

Comment (anastasiia-kornilova) [reply src/algorithmForNN/fileUtils.py]: Looks like smth like `labels_list[np.array(index_list) + 1] = "1,"` should work. Also, I don't like the idea to keep labels as strings. Let's separate logic on label values and logic on storage format (that this should be csv string).

**Review by anastasiia-kornilova — CHANGES_REQUESTED** [review]

**Review by anastasiia-kornilova — APPROVED** [review]

**Pull request merged**